lib/Tooling/Refactor/Extract.cpp

```diff
@@ -171,8 +184,12 @@ Optional<CompoundStatementRange> getExtractedStatements(const CompoundStmt *CS,
   assert(Begin && End);
   CompoundStatementRange Result;
   Result.First = std::find(CS->body_begin(), CS->body_end(), Begin);
+  if (Result.First == CS->body_end())
+    Result.First = comprehensiveFind(CS, Begin);
   assert(Result.First != CS->body_end());
   Result.Last = std::find(Result.First, CS->body_end(), End);
+  if (Result.Last == CS->body_end())
+    Result.Last = comprehensiveFind(CS, End);
```

Comment:
This comment won't apply if you plan on reusing comprehensiveFind elsewhere, but you could avoid some pointer chasing by calling comprehensiveFind once and looking for both Begin & End?